Include/internal/pycore_cell.h

```diff
@@ -18,8 +19,8 @@ PyCell_SwapTakeRef(PyCellObject *cell, PyObject *value)
 {
     PyObject *old_value;
     Py_BEGIN_CRITICAL_SECTION(cell);
-    old_value = cell->ob_ref;
-    cell->ob_ref = value;
+    FT_ATOMIC_STORE_PTR_RELEASE(old_value, cell->ob_ref);
+    FT_ATOMIC_STORE_PTR_RELEASE(cell->ob_ref, value);
```

Comment:
This is a little suspicious. Why do we need the atomics when there's the critical section?

Reply:
```
SUMMARY: ThreadSanitizer: data race /home/runner/work/cpython/cpython/./Include/internal/pycore_cell.h:23:18 in PyCell_SwapTakeRef
```

Reply:
We need the second atomic write here (to `cell->ob_ref`), but not the first one. In other words, something like:

```c
old_value = cell->ob_ref;
FT_ATOMIC_STORE_PTR_RELEASE(cell->ob_ref, value);
```

Although the write happens in the critical section, the read happens (with this PR) outside of the lock.  This is the same pattern we use in dictionaries :

https://github.com/python/cpython/blob/da5855e99a8c2d6ef2bb20124d2ebb862dbb971f/Objects/dictobject.c#L278-L279